From 2948110acbfa1c28a1072d97dcc03175ddd15f90 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:00:14 -0400
Subject: [acpi] dock: Do not check CONFIG_ACPI_DOCK_MODULE

Message-id: <20140710010014.22146.53844.stgit@gir.stowe>
Patchwork-id: 85558
O-Subject: [RHEL7.1 PATCH 019/105] ACPI / dock: Do not check CONFIG_ACPI_DOCK_MODULE
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: e6c215f15a1cbf4cdd6996d95fba0c0d6c0f3ccc

commit e6c215f15a1cbf4cdd6996d95fba0c0d6c0f3ccc
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Fri Jul 5 03:02:25 2013 +0200

    ACPI / dock: Do not check CONFIG_ACPI_DOCK_MODULE

    Since commit 94add0f (ACPI / dock: Initialize ACPI dock subsystem
    upfront) the ACPI dock driver cannot be a module, so
    CONFIG_ACPI_DOCK_MODULE is never set.  For this reason, simplify
    the preprocessor conditional in include/acpi/acpi_drivers.h
    referring to that sybbol unnecessarily.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/acpi/acpi_drivers.h b/include/acpi/acpi_drivers.h
index b420939..0cf8578 100644
--- a/include/acpi/acpi_drivers.h
+++ b/include/acpi/acpi_drivers.h
@@ -117,7 +117,7 @@ struct acpi_dock_ops {
  acpi_notify_handler uevent;
 };
 
-#if defined(CONFIG_ACPI_DOCK) || defined(CONFIG_ACPI_DOCK_MODULE)
+#ifdef CONFIG_ACPI_DOCK
 extern int is_dock_device(acpi_handle handle);
 extern int register_dock_notifier(struct notifier_block *nb);
 extern void unregister_dock_notifier(struct notifier_block *nb);
@@ -150,6 +150,6 @@ static inline int register_hotplug_dock_device(acpi_handle handle,
 static inline void unregister_hotplug_dock_device(acpi_handle handle)
 {
 }
-#endif
+#endif /* CONFIG_ACPI_DOCK */
 
 #endif /*__ACPI_DRIVERS_H__*/
-- 
1.7.1